### DISCUSSION TITLE
Whisper not found, but it's installed

### DISCUSSION THREAD

**Discussion by Pivchanskij** [SubtitleEdit/subtitleedit#6328] [Q&A] [answered]
I installed whisper, but Subtitle Edit doesn't see it. Try to add variable, no results. How do I fix this?
![pip list](https://user-images.githubusercontent.com/105757697/195735290-68717b37-1638-4043-88bc-e8081f79d7b0.jpg)
![not found](https://user-images.githubusercontent.com/105757697/195735328-fe6880db-f5e4-4f4a-a019-32c0501c5bde.jpg)
![variable](https://user-images.githubusercontent.com/105757697/195735334-d9f1a85b-993d-449c-9c52-abd4a625f16e.jpg)

**Comment by niksedk** [comment, marked as answer]
Thx - is latest beta better? https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip

**Reply by Pivchanskij** [reply]
Yes, thx

**Reply by Pivchanskij** [reply]
There is a new problem. After recognition, an empty .srt file is created. Tried Russian and English. And several models: large, base, base.en.
![log](https://user-images.githubusercontent.com/105757697/195835325-c4f62a0c-856f-442f-84f7-9537bb281c4f.jpg)

**Reply by Pivchanskij** [reply]
Solved the problem. Added a environment variable where ffmpeg is located.

**Reply by niksedk** [reply]
OK, latest beta should not need an environment variable for ffmpeg, if ffmpeg is activated in SE (Option - Settings - Waveform).

**Comment by coastal45** [comment]
I have similar problem "Whisper not found." (beta version from 10/14) In Windows I installed Whisper by `pip3 install whisper` rather than git method.
Whisper installed properly, but the files location is different from the example above.
They are in: \AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\Scripts

Is there any way SE can find them?

**Reply by niksedk** [reply]
OK, I've tried to support this too... and SE now has a setting in Settings.xml called "WhisperLocation", so you can put the "whisper.exe" file anywhere you want ;)

https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip

**Comment by coastal45** [comment]
@niksedk Finally I got Whisper working on Windows 10. I think `pip3 install whisper` was a mistake.
So I started over.

install python
https://www.python.org/downloads/release/python-3108/

install pip
https://phoenixnap.com/kb/install-pip-windows

install git
https://phoenixnap.com/kb/how-to-install-git-windows

install whisper
At command prompt: `pip install git+https://github.com/openai/whisper.git`

then latest SE beta found whisper. And models loaded ok. But in operation, I got a "No text found" error dialog.

![Screenshot 2022-10-16 123728a](https://user-images.githubusercontent.com/102390147/196055929-6abbb3bc-8656-4e46-af6b-c7ebf77f6354.jpg)
 
I guess it's the same issue as in https://github.com/SubtitleEdit/subtitleedit/discussions/6300

In the F2 log:

Calling whisper with : whisper --model tiny --language "Japanese" --fp16 False "C:\Users\robert2\AppData\Local\Temp\831260ff-2bee-4519-a1a2-d5f8d87528ff.wav"
Wav file name: C:\Users\robert2\AppData\Local\Temp\831260ff-2bee-4519-a1a2-d5f8d87528ff.wav

Any update on this?

**Comment by coastal45** [comment]
@niksedk  Checking further, setting the language to English or French returned text, as attached (of course it's wrong...).
But selecting Japanese or Chinese got the "No text found!" dialog. Maybe it's a Whisper installation issue in Windows?
On Linux it works correctly though. 

[srt files.zip](https://github.com/SubtitleEdit/subtitleedit/files/9863214/srt.files.zip)

**Comment by coastal45** [comment]
@niksedk Thanks again, however I still can't get past this on Windows 10. And advice or what I could try?
Currently, release version 3.6.8 still has this problem. I removed --fp32 from settings.xml (someone else mentioned that) but no change.

**Comment by coastal45** [comment]
@niksedk   Regarding the "No text found!" error in SE on Windows, along with:
`UnicodeEncodeError: 'charmap' codec can't encode characters in position 26-37: character maps to <undefined>`
in the error log, could this be related to the fact that Command Prompt in my system does not display non-ASCII characters?

I checked on how to make Command Prompt display UTF-8 in Windows 10, but it gets into regedit and system mods. that I don't understand well enough to trust.
Perhaps you know a way to do this? It might solve this problem.

**Reply by ard-av** [reply]
v 3.6.10? as I understand it, there is another mechanism (cpp)
![Screenshot_1](https://user-images.githubusercontent.com/114281218/205005153-05be8729-f70e-407d-882a-faaa667335d0.png)
In other versions, wrote Text not found. In this version, speech-to-text translation worked without installing python and other settings. Thanks to the developer. Great job

**Reply by coastal45** [reply]
I couldn't get cpp to work on Windows at all, but I think it's a different problem.
So I want to stay with php for now. Maybe I'll start a separate issue on this.
